src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -1198,6 +1196,26 @@ void network::execute_impl(const std::vector<event::ptr>& events) {
 
     set_arguments();
     GPU_DEBUG_GET_INSTANCE(debug_config);
+    // Load binary dump for input layers
+    GPU_DEBUG_IF(!debug_config->load_layers_raw_dump.empty()) {
+        for (auto& inst : _exec_order) {
+            const std::string layer_name = inst->id();
+            GPU_DEBUG_IF(debug_config->is_loading_layer(layer_name)) {
+                auto dump_file = debug_config->get_loading_name(layer_name);
+                GPU_DEBUG_COUT  << " Load binary dump : " << dump_file << " for " << layer_name << std::endl;
+                if (dump_file.length() > 0) {
+                    std::vector<uint8_t> bin = ov::util::load_binary(dump_file);
+                    OPENVINO_ASSERT(!bin.empty(), "Failure loading binary from OV_GPU_LoadDumpRawBinary : " + dump_file);
+
+                    auto input_mem = get_primitive(layer_name)->output_memory_ptr(0);
+                    OPENVINO_ASSERT(input_mem->size() == bin.size(), "memory size mis-match for OV_GPU_LoadDumpRawBinary : " + layer_name);
+
+                    input_mem->copy_from(get_stream(), static_cast<void *>(&bin[0]), true);
+                }
+            }
+        }
+    }
```

Comment:
this works only for the input of network. Can we set data for any layer?

Reply:
moved into loop.